mkt/websites/views.py

```diff
@@ -61,3 +70,200 @@ class ReviewersWebsiteSearchView(WebsiteSearchView):
                        SortingFilter]
     serializer_class = ReviewerESWebsiteSerializer
     form_class = ReviewersWebsiteSearchForm
+
+
+class WebsiteMetadataScraperView(CORSMixin, MarketplaceView, APIView):
+    """
+    Base website search view based on a single-string query.
+    """
+    cors_allowed_methods = ['get']
+    authentication_classes = [RestSharedSecretAuthentication,
+                              RestOAuthAuthentication]
+    permission_classes = [AllowAny]
+
+    errors = {
+        'no_url': '`url` querystring parameter required',
+        'network': 'Unable to fetch website metadata',
+        'malformed_data': 'Unable to parse website HTML.'
+    }
+
+    def get_name(self, doc):
```

Comment:
I'd rather see these methods pulled out of the view also since they don't have anything to do with a request and it seems like something we might want to use elsewhere.